storage-proofs/src/vde.rs

```diff
@@ -126,3 +127,26 @@ pub fn create_key<H: Hasher>(
     let hash = hasher.finalize();
     Ok(bytes_into_fr_repr_safe(hash.as_ref()).into())
 }
+
+/// Creates the encoding key from a `MerkleTree`.
+/// The algorithm for that is `Blake2s(id | encodedParentNode1 | encodedParentNode1 | ...)`.
+/// It is only public so that it can be used for benchmarking
+pub fn create_key_from_tree<H: Hasher>(
```

Comment:
It would be great to have a test to ensure the two `crate_key` impls don't get out of sync

Reply:
I don't know how to do that, the previous "check this" `FIXME` note drew attention to this function because I basically copy-pasted changing slices to trees, I don't even like having both functions.